src/expression.js

```diff
@@ -781,13 +787,22 @@ var expression = {
 						key: token.slice(1) // remove leading `.`
 					});
 				}
-				else if(firstParent.type === 'Bracket' && !(firstParent.children && firstParent.children.length > 0)) {
-					stack.addTo(["Bracket"], {type: "Lookup", key: token});
-				}
-				// This is to make sure in a helper like `helper foo[bar] car` that
-				// car would not be added to the Bracket expression.
-				else if(stack.first(["Helper", "Call", "Hash","Arg"]).type === 'Helper') {
-					stack.addToAndPush(["Helper"], {type: "Lookup", key: token});
+				else if(firstParent.type === 'Bracket') {
+					if (!(firstParent.children && firstParent.children.length > 0)) {
```

Comment:
It would be nice to add some inline docs here similar to those below so we know what this `if` is "really" testing.